qiskit/algorithms/minimum_eigen_solvers/adapt_vqe.py

```diff
@@ -0,0 +1,338 @@
+    ):
+        if logger.isEnabledFor(logging.INFO):
+            gradlog = []
+            gradlog.append(
+                f"\nGradients in iteration #{str(iteration)}\n"
+                "ID: Excitation Operator: Gradient  <(*) maximum>"
+            )
+            for i, grad in enumerate(cur_grads):
+                gradlog.append(f"\n{str(i)}: {str(grad[1])}: {str(grad[0])}")
+                if grad[1] == max_grad[1]:
+                    gradlog.append("\t(*)")
+            logger.info(",".join(gradlog))
+
+    def compute_minimum_eigenvalue(
+        self,
+        operator: OperatorBase,
+        aux_operators: Optional[ListOrDict[PauliSumOp]] = None,
+    ) -> AdaptVQEResult:
+        """Computes the minimum eigenvalue.
+
+        Args:
+            operator: Operator to evaluate.
+            aux_operators: Additional auxiliary operators to evaluate.
+
+        Raises:
+            QiskitError: If a solver other than VQE
+                (`qiskit.algorithms.minimum_eigen_solvers.vqe.py`) or a ansatz other than
+                EvolvedOperatorAnsatz (`qiskit.circuit.library.evolved_operator_ansatz.py`)
+                is provided or if the algorithm finishes due to an unforeseen reason.
+            AlgorithmError: If `quantum_instance` is not provided.
+            QiskitError: If the user-provided `aux_operators` contain a name which clashes
+                with an internally constructed auxiliary operator. Note: the names used for the
+                internal auxiliary operators correspond to the `Property.name` attributes which
+                generated the respective operators.
+            QiskitError: If the chosen gradient method appears to result in all-zero gradients.
+
+        Returns:
+            An AdaptVQEResult (`qiskit.algorithms.minimum_eigen_solvers.adapt_vqe.AdaptVQEResult`)
+            which is a VQEResult (`qiskit.algorithms.minimum_eigen_solvers.vqe.VQEResult`) but also
+            includes runtime information about the AdaptVQE algorithm like the number of iterations,
+            termination criterion, and the final maximum gradient.
+        """
+        if not isinstance(self._tmp_ansatz, EvolvedOperatorAnsatz):
+            raise QiskitError("The AdaptVQE ansatz must be of the EvolvedOperatorAnsatz type.")
+
+        if self._solver.quantum_instance is None:
+            raise AlgorithmError(
+                "A QuantumInstance or Backend must be supplied to run the quantum algorithm."
+            )
+
+        # construct the expectation
+        if self._solver.expectation is None:
+            expectation = ExpectationFactory.build(
+                operator=operator,
+                backend=self._solver.quantum_instance,
+                include_custom=self._solver.include_custom,
+            )
+        else:
+            expectation = self._solver.expectation
+
+        # Overwrite the solver's ansatz with the initial state
+        self._solver.ansatz = self._tmp_ansatz.initial_state
```

Comment:
I think we should be careful not to change the state of the `VQE` we hold internally, since the user might re-use the same VQE instance later on like:
```python
vqe = VQE(..)
adapt_vqe = AdaptVQE(vqe, ..)
adapt_result = adapt_vqe.compute_minimum_eigenvalue(..)

# do something else with vqe
```

This could also be done by restoring the ansatz to the original value at the end.